src/Patch/Document.js

```diff
@@ -0,0 +1,67 @@
+import Native from './Native';
+import CustomElementInternals from '../CustomElementInternals';
+import * as Utilities from '../Utilities';
+
+import PatchParentNode from './Interface/ParentNode';
+
+/**
+ * @param {!CustomElementInternals} internals
+ */
+export default function(internals) {
+  Utilities.setPropertyUnchecked(Document.prototype, 'createElement',
+    /**
+     * @this {Document}
+     * @param {string} localName
+     * @return {!Element}
+     */
+    function(localName) {
+      const definition = internals.localNameToDefinition(localName);
+      if (definition) {
+        return new (definition.constructor)();
+      }
+
+      const result = /** @type {!Element} */
+        (Native.Document_createElement.call(this, localName));
+      internals.patch(result);
+      return result;
+    });
+
+  Utilities.setPropertyUnchecked(Document.prototype, 'importNode',
+    /**
+     * @this {Document}
+     * @param {!Node} node
+     * @param {boolean=} deep
+     * @return {!Node}
+     */
+    function(node, deep) {
+      const clone = Native.Document_importNode.call(this, node, deep);
+      internals.patchTree(clone);
+      internals.upgradeTree(clone);
+      return clone;
+    });
+
+  const NS_HTML = "http://www.w3.org/1999/xhtml";
+
+  Utilities.setPropertyUnchecked(Document.prototype, 'createElementNS',
+    /**
+     * @this {Document}
+     * @param {?string} namespace
+     * @param {string} localName
+     * @return {!Element}
+     */
+    function(namespace, localName) {
+      if (namespace === null || namespace === NS_HTML) {
+        return this.createElement(localName);
+      }
+
+      const result = /** @type {!Element} */
+        (Native.Document_createElementNS.call(this, namespace, localName));
+      internals.patch(result);
```

Comment:
Probably not necessary? I'm not sure `innerHTML` is even well defined enough to know what should happen when `innerHTML` is read/written on elements in other namespaces. However, given that `innerHTML` is on the Element prototype - as opposed to HTMLElement - I think patching this would be more correct.